lib/util/browserdetection.js

```diff
@@ -17,6 +17,28 @@ function isIOS() {
   return /iPad|iPhone|iPod/.test(navigator.userAgent);
 }
 
+/**
+ * Detects whether or not a device is an iPad
+ * @returns {boolean}
+ */
+function isIpad() {
+  return navigator && navigator.maxTouchPoints &&
+  navigator.maxTouchPoints > 2 &&
+  window.screen.width >=  768 &&
+  (/Macintosh/i.test(navigator.userAgent) || /iPad/.test(navigator.userAgent) || /iPad/.test(navigator.platform));
+}
+
+/**
+ * Detects whether or not a device is an iPhone
+ * @returns {boolean}
+ */
+function isIphone() {
+  return navigator && navigator.maxTouchPoints &&
+  navigator.maxTouchPoints > 2 &&
+  window.screen.width <=  476 &&
```

Comment:
just curious - does the width stay at `window.screen.width <=  476` when iPhone is rotated? same question for iPad

Reply:
Yes. Just tested this on my iPhone and it's the same value!

Reply:
I used the largest width from iPhones (iPhone 7 Plus was the last of this size!) and the smallest width from iPads (Starting from the "smallest" iPads which were 768 in width). 

However, I can go one bit smaller and that this helped me realize that iPad minis (744 width) which are no longer in production have a slightly smaller screen size but I'll include it as part of this.

Otherwise, there doesn't seem to be any devices currently available from Apple that have overlaps in widths for these devices.